app/workers/facilities/pssg_download.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require 'rgeo/geo_json'
+
+module Facilities
+  class PSSGDownload
+    include Sidekiq::Worker
+    include SentryLogging
+
+    def perform
+      @drivetime_band_client = Facilities::DrivetimeBandClient.new
+      download_data
+    end
+
+    private
+
+    def create_and_save_drive_time_data(drive_time_data)
+      attributes = drive_time_data.dig('attributes')
+
+      vha_id = extract_vha_id(attributes)
+      facility = BaseFacility.find_facility_by_id(vha_id)
+      return if facility.nil?
+
+      drive_time_band = facility.drivetime_bands.find_or_initialize_by(vha_facility_id: extract_name(attributes))
+      drive_time_band.min = attributes.dig('FromBreak')
+      drive_time_band.max = attributes.dig('ToBreak')
+      drive_time_band.name = attributes.dig('Name')
+      drive_time_band.polygon = extract_polygon(drive_time_data)
+      facility.save
+    end
+
+    def extract_name(attributes)
+      name = attributes.dig('Name')
+      name.partition(':').first.strip!
+    end
+
+    def extract_vha_id(attributes)
+      'vha_' + extract_name(attributes)
+    end
```

Comment:
You could get rid of `def extract_vha_id` and reduce the calls to `extract_name` from 2 down to 1 with something like this:
```ruby
name = extract_name(attributes)
vha_id = 'vha_' + name
#...
drive_time_band = facility.drivetime_bands.find_or_initialize_by(vha_facility_id: name)
```

Reply:
Thanks, I added some exception handling but not sure if it's pragmatically correct in ruby